### DISCUSSION TITLE
Troubleshooting Lottie Integration Errors in GitHub Actions CI/CD Pipeline

### DISCUSSION THREAD

**Discussion by mBellar** [airbnb/lottie-ios#2332] [Q&A] [answered]
Hello Everyone, 

I am encountering difficulties with integrating Lottie into our CI/CD pipeline on GitHub Actions. I am using the lottie-spm extension. When running our workflow, I encounter errors indicating that it cannot find the 'LottieAnimation' and 'LottieAnimationView' types within the scope. These errors only occur on machines hosted by GitHub Actions, while execution on our local machines proceeds without any issues.

![Screenshot 2024-03-11 at 10 28 30](https://github.com/airbnb/lottie-ios/assets/147486707/ae9842a5-1388-4487-9048-6fa867e9c9cc)

However, when attempting to compile the application using SDK 'iphoneos17.2' with Fastlane, I encounter a similar error:

![Screenshot 2024-03-11 at 12 03 42](https://github.com/airbnb/lottie-ios/assets/147486707/0cf3546d-6f8e-45f4-9b5b-d75663ad7011)

Then, upon changing the SDK to 'iphoneos17.0', I received this error:

`xcodebuild: error: SDK "iphoneos17.0" could not be found.`

My GitHub Actions workflow is currently configured to run on macOS 13.
I use as environment : 
- Xcode 15.0.1
- SDK Name : iphoneos17.2.

Could you help us resolve these errors related to Lottie and understand how to properly specify the SDK in Fastlane to solve the compatibility issue?

Thank you very much for your help.

Best regards

**Comment by calda** [comment]
I haven't seen this issue before and don't know what could cause it. I could take a look if you share a sample project that reproduces this error.

Have you tried using `lottie-ios` instead of `lottie-spm`?

**Reply by imrmaximus** [reply]
Hi @calda !
We use `lottie-spm` through another lib so we cannot decide how `lottie` is loaded.

**Comment by mBellar** [comment]
The issue lies within our CI/CD pipeline setup rather than the Lottie integration itself. When using lottie-spm, the build job fails due to the inability to locate 'LottieAnimation' and 'LottieAnimationView' types. However, switching to lottie-ios resolves the issue.

Unfortunately, I couldn't reproduce this error in other projects locally, indicating it's likely specific to our CI/CD environment. It's crucial for us to use lottie-spm due to an external SDK dependency.
 Any insights or suggestions on how to resolve this would be greatly appreciated.

**Reply by calda** [reply]
You could try testing using different Xcode versions / different GitHub actions runners. You can configure the Xcode version your job uses by running `sudo xcode-select --switch /Applications/Xcode_VERSION.app` in your action, and could test on the new [macOS 14 runner](https://github.com/actions/runner-images/blob/main/images/macos/macos-14-arm64-Readme.md).

**Comment by rwarrender** [comment]
Have you tried manually downloading the xcframework and dragging it into the project instead?

You can download it from https://github.com/airbnb/lottie-ios/releases/download/4.4.1/Lottie-Xcode-15.2.xcframework.zip

**Reply by mBellar** [reply]
However, I actually need to use the Lottie SPM rather than Lottie iOS. Do you happen to have any other suggestion ?

**Reply by imrmaximus** [reply]
Hi Richard,
I work with @mBellar on this subject.
In fact, the problem is that we load lottie-spm through another library, so don't really control on how lottie-spm is imported.

**Reply by imrmaximus** [reply]
In our last tests, we load the framework directly in the xcode project, or through a package (the other library we load (which loads the lottie-spm package) is loaded as dependency of another inner package) and we still have the same errors:
`cannot find type 'LottieAnimation' in scope`

**Reply by rwarrender** [reply]
Does the issue still appear if you archive locally?

**Reply by imrmaximus** [reply]
Hi @rwarrender ,
Thanks for your suggestion but, good news, we fix our problem. 😄

**Comment by imrmaximus** [comment, marked as answer]
Hi todos,
We solved our issue.
The problem came from our use of the fastlane build_app. We didn't precise that the system must clean the compile code before execution, so it used the old version of the lottie library with our code (instead of using the `lottie-spm`).

So, now, all is fine and thank you for your helps.

**Reply by calda** [reply]
Excellent! For future reference, what code / configuration change did you have to make to fix the issue?

**Reply by imrmaximus** [reply]
I just added the `clean: true` as parameter in `build_app` call of our fastlane script.
